src/lsp_data.rs

```diff
@@ -85,11 +86,18 @@ pub mod ls_util {
         }
     }
 
-    pub fn range_from_vfs_file(_vfs: &Vfs, _fname: &Path) -> Range {
-        // FIXME: todo, endpos must be the end of the document, this is not correct
-
-        let end_pos = Position::new(0, 0);
-        Range{ start: Position::new(0, 0), end: end_pos }
+    pub fn range_from_vfs_file(_vfs: &Vfs, _fname: &Path) -> Result<Range, vfs::Error> {
```

Comment:
The reason why I just returned the error is, that this function cannot be sure if the `Vfs` will error or not. Thus calling this function with the wrong parameters would unnecessarily panic. However, the calling code can know that this file must exists, e.g., because the client transmitted the filename to us.